crates/pep440-rs/src/version.rs

```diff
@@ -2320,11 +2412,32 @@ pub(crate) fn compare_release(this: &[u64], other: &[u64]) -> Ordering {
 ///
 /// [pep440-suffix-ordering]: https://peps.python.org/pep-0440/#summary-of-permitted-suffixes-and-relative-ordering
 fn sortable_tuple(version: &Version) -> (u64, u64, Option<u64>, u64, &[LocalSegment]) {
-    match (version.pre(), version.post(), version.dev(), version.min()) {
+    // If the version is a "max" version, use a post version larger than any possible post version.
+    let post = if version.is_max() {
+        // But don't convert a `dev` version to a `post` version. `1.0.dev0` should _not_ be greater
+        // than `1.0.post0.dev0`, since the latter means "a dev release of a post release of 1.0",
+        // not "a post release of a dev release of 1.0".
```

Comment:
Mildly concerned with how all of this intersects with post-releases, but I did add tests.

Reply:
I ended up removing `dev` support, we never create a version with both `max` and `dev` so not sure it's worth accommodating it here.

Reply:
Is this caveat documented?

Reply:
I added debug asserts to the constructors for `min` and `max`.